packages/typespec-autorest/src/emit.ts

```diff
@@ -98,7 +98,7 @@ export async function getAllServicesAtAllVersions(
   program: Program,
   options: ResolvedAutorestEmitterOptions
 ): Promise<AutorestServiceRecord[]> {
-  const tcgcSdkContext = createSdkContext(
+  const tcgcSdkContext = await createSdkContext(
```

Comment:
won't this be breaking for all language emitters because we now have to add async? It's a small break, but I think we should call it out as such

Reply:
yes. it will break all emitters. also, i don't think there is a good way to work around this breaking. i will let all impacted emitter be aware of this before merge.

Reply:
Yeah, java calls it in `constructor` :-(

Reply:
will it be a problem for java to change to async?

Reply:
Probably not blocking problem. Just need to make some (non-trivial) refactor.